docs/getting-started-guides/fedora/flannel_multi_node_cluster.md

```diff
@@ -0,0 +1,159 @@
+
+This document describes how to deploy kubernetes on multiple hosts to set up a multi-node cluster and networking with flannel. Follow fedora [getting started guide](fedora_manual_config.md) to setup 1 master (fed-master) and 2 or more nodes (minions). Make sure that all nodes (minions) have different names (fed-node1, fed-node2 and so on) and labels (fed-node1-label, fed-node2-label, and so on) to avoid any conflict. Also make sure that the kubernetes master host is running etcd, kube-controller-manager, kube-scheduler, and kube-apiserver services, and the nodes (minions) are running docker, kube-proxy and kubelet services. Now install flannel on kubernetes nodes (minions). flannel on each node configures an overlay network that docker uses. flannel runs on each node to setup a unique class-C container network.
+
+##**Perform following commands on the kubernetes master**
+
+* Configure flannel by creating a `flannel-config.json` in your current directory on fed-master. flannel provides udp and vxlan among other overlay networking backend options. In this guide, we choose kernel based vxlan backend. The contents of the json are:
+
+```
+{
+    "Network": "18.16.0.0/16",
+    "SubnetLen": 24,
+    "Backend": {
+        "Type": "vxlan",
+        "VNI": 1
+     }
+}
+```
+**NOTE:** Choose an IP range that is *NOT* part of the public IP address range.
+
+* Add the configuration to the etcd server on fed-master.
+
+```
+# etcdctl set /coreos.com/network/config < flannel-config.json
+```
+
+* Verify the key exists in the etcd server on fed-master.
+
+```
+# etcdctl get /coreos.com/network/config
+```
+
+##**Perform following commands on all kubernetes nodes**
+
+* Edit the flannel configuration file /etc/sysconfig/flanneld as follows:
+
+```
+# Flanneld configuration options
+
+# etcd url location.  Point this to the server where etcd runs
+FLANNEL_ETCD="http://fed-master:4001"
+
+# etcd config key.  This is the configuration key that flannel queries
+# For address range assignment
+FLANNEL_ETCD_KEY="/coreos.com/network"
+
+# Any additional options that you want to pass
+FLANNEL_OPTIONS=""
+```
+
+**Note:** By default, flannel uses the interface for the default route. If you have multiple interfaces and would like to use an interface other than the default route one, you could add "-iface=" to FLANNEL_OPTIONS. For additional options, run `flanneld --help` on command line.
+
+* Enable the flannel service.
+
+```
+# systemctl enable flanneld
+```
+
+* If docker is not running, then starting flannel service is enough and skip the next step.
+
+```
+# systemctl start flanneld
+```
+
+* If docker is already running, then stop docker, delete docker bridge (docker0), start flanneld and restart docker as follows. Another alternative is to just reboot the system (`systemctl reboot`).
+
+```
+# systemctl stop docker
+# ip link delete docker0
+# systemctl start flanneld
+# systemctl start docker
+```
+
```

Comment:
Should we move the "Test the cluster and flannel configuration" section here?  I really don't want this to seem like it is hard to set up   :)